shell/plugins/plugin.js

```diff
@@ -26,38 +27,69 @@ export default async function(context) {
   }
 
   if (loadPlugins) {
-    // TODO: Get rancher version using the new API (can't use setting as we have not loading the store)
-    const rancherVersion = undefined;
+    let rancherVersion;
+    let kubeVersion;
+
+    const reqs = [];
+
+    // Fetch rancher version metadata
+    reqs.push(context.store.dispatch('rancher/request', {
+      url:                  '/rancherversion',
```

Comment:
The problem is that the loading of the extensions is unaware of the promise/await in the intialisation of the `version` vue plugin (file that I deleted).

They are executed "kinda" simultaneously (check here: https://github.com/rancher/dashboard/blob/master/shell/initialize/install-plugins.js#L46-L57) and `rancherVersion` will never get populated in the context of `shell/plugins/plugin.js`.

I've looked into this and I didn't see any kind of problem, since the data for `/rancherversion` is populating a JS variable for propagation/usage and it's network request is "kinda" done at the same time, still.

Worth to note that the JS variable is still being populated though, so transmission of data should be assured.

Is there any other problem you see with this that I am not aware? Because the way it was being done (which I kept) didn't not assure that the data was there at a given point in time because of the data awareness. Wdyt?

Reply:
If `loadPlugins` is false we never make the request, so anything that depends on rancherversion will be broken

Reply:
That's true @richard-cox . Forgot about that. Let me adjust the script to always do the `/rancherversion` network request 🙏

Reply:
PR updated